### PR TITLE
chore(repo): Speed up PR title linter

### PR DESCRIPTION
**Pull request by tmilewski**
## Description

Remove unnecessary tasks from the PR title linter.

`~2m15s => ~18s`

Fixes SDK-1803

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: .github/workflows/pr-title-linter.yml [hunk 1]
```diff
@@ -16,15 +16,16 @@ jobs:
...
-
-      - name: Setup
-        id: config
-        uses: ./.github/actions/init
         with:
-          turbo-signature: ${{ secrets.TURBO_REMOTE_CACHE_SIGNATURE_KEY }}
-          turbo-team: ${{ vars.TURBO_TEAM }}
-          turbo-token: ${{ secrets.TURBO_TOKEN }}
-          playwright-enabled: true # Must be present to enable caching on branched workflows
+          fetch-depth: 1
```
Comment (LauraBeatris) [inline .github/workflows/pr-title-linter.yml:20]: The default value for `fetch-depth` is 1. We could remove it, or leave it to be explicit.

File: .github/workflows/pr-title-linter.yml [hunk 2]
```diff
@@ -16,15 +16,16 @@ jobs:
...
+          sparse-checkout: |
+            commitlint.config.ts
+            packages/*/package.json
+          sparse-checkout-cone-mode: false
 
       - name: Lint Pull Request Title
-        run: echo "${{ github.event.pull_request.title }}" | npx commitlint
+        run: |
+          npm init --scope=clerk --yes
+          npm i globby commitlint --audit=false --fund=false
```
Comment (LauraBeatris) [inline .github/workflows/pr-title-linter.yml:30]: What's the usage of `globby` here?

**Review by LauraBeatris — APPROVED** [review]

File: .github/workflows/pr-title-linter.yml (hunk 1 above)
Comment (tmilewski) [reply .github/workflows/pr-title-linter.yml:20]: You're correct, the docs have the a `0` right next to the `1`, and I read that as the default.

File: .github/workflows/pr-title-linter.yml (hunk 2 above)
Comment (tmilewski) [reply .github/workflows/pr-title-linter.yml]: Globby is being used in the Commitlint config file itself.

File: .github/workflows/pr-title-linter.yml [hunk 3]
```diff
@@ -16,15 +16,17 @@ jobs:
...
+            commitlint.config.ts
+            packages/*/package.json
+            tsconfig.json
+          sparse-checkout-cone-mode: false
 
       - name: Lint Pull Request Title
-        run: echo "${{ github.event.pull_request.title }}" | npx commitlint
+        run: |
+          npm init --scope=clerk --yes
+          npm i --save-dev @commitlint/cli globby typescript --audit=false --fund=false
```
Comment (BRKalow) [inline .github/workflows/pr-title-linter.yml:31]: Can we add a comment here and/or in the commitlint config noting the dependency between the packages installed here and those used in our commitlint config? If the commitlint is updated in some way we need to make sure the packages it relies on are installed in this workflow.

**Pull request merged**